### DISCUSSION TITLE
Cypress component testing(react/angular) vs karma/jasmine vs jest

### DISCUSSION THREAD

**Discussion by BrkCoder** [cypress-io/cypress#17239] [Component Testing] [answered]
In these current days we are integrating cypress as our main e2e framework. 
I would like to know what are the benefits of integrating it(cypress) also into our unit tests ?
We currently develop an angular app.

1. What are the advantages & disadvantages of cypress vs karma/jasmine vs jest
2. Does cypress component testing framework is faster then karma test runner & jest ?
3. Does the migration can be painful , we have thousands of test units written in jasmine/karma ?

Thanks in advance !

**Comment by bahmutov** [comment, marked as answer]
Read my vision for component testing here https://glebbahmutov.com/blog/my-vision-for-component-tests/

In my opinion, component testing in a real browser using Cypress are 10x better than writing tests in karma/jasmine and jest

1. real browser compared to jest, much easier debugging experience
2. same speed as jest. Only the initial Cypress startup time is longer
3. use any plugin like visual testing, reporter, etc is available
4. network stubbing and mocking using cy.intercept command

I don't think you need to rewrite the tests, but I would try writing a few component tests and see what you think

**Reply by BrkCoder** [reply]
thanks you so much for your quick answers.
a final question, from what I have read I know that angular team work closely with cypress 's team for Component test harnesses(aka [https://material.angular.io/cdk/test-harnesses/overview](https://material.angular.io/cdk/test-harnesses/overview)), are you going to add component test harnesses in the future like in angular material?

**Reply by saivishnutammineni** [reply]
This answers seems to be comparing jest and cypress. What are the advantages over jasmine karma tests? Speed?
